src/lib/core/fiber.c

```diff
@@ -258,11 +258,19 @@ pthread_t main_thread_id;
 static size_t page_size;
 static int stack_direction;
 
+#if FIBER_STACK_SIZE < 524288
+/*
+ * See the rationale for the minimal fiber stack size in
+ * https://github.com/tarantool/tarantool/issues/3418.
+ */
+#error "Minimal fiber stack size is 512Kb"
+#endif
+
```

Comment:
Nit: It'd probably be better to perform this check in cmake instead of postponing it till compile time.

Reply:
I had some doubts regarding this decision: I still think, that we must have such kind of compile time check to avoid incompatibility between source code and CMake machinery, but I agree, that diagnostics should be moved to CMake. Hence, the changes:

```diff --git a/cmake/SetFiberStackSize.cmake b/cmake/SetFiberStackSize.cmake
index 2726cac7d..58aa7f751 100644
--- a/cmake/SetFiberStackSize.cmake
+++ b/cmake/SetFiberStackSize.cmake
@@ -27,8 +27,15 @@ endif()
 math(EXPR FIBER_STACK_SIZE_IN_BYTES
     "((${FIBER_STACK_SIZE_IN_BYTES} >> 12) + 1) << 12")
 
-message(STATUS "[SetFiberStackSize] Fiber stack size: ${FIBER_STACK_SIZE}"
-    " (adjusted to ${FIBER_STACK_SIZE_IN_BYTES} bytes)")
+# See the rationale for the minimal fiber stack size in
+# https://github.com/tarantool/tarantool/issues/3418.
+if(FIBER_STACK_SIZE_IN_BYTES LESS 524288)
+    message(FATAL_ERROR "[SetFiberStackSize] Minimal fiber stack size is 512Kb,"
+        " but ${FIBER_STACK_SIZE} is given")
+else()
+    message(STATUS "[SetFiberStackSize] Fiber stack size: ${FIBER_STACK_SIZE}"
+        " (adjusted to ${FIBER_STACK_SIZE_IN_BYTES} bytes)")
+endif()
 
 # Propagate the value to the sources.
 add_definitions(-DFIBER_STACK_SIZE=${FIBER_STACK_SIZE_IN_BYTES})
diff --git a/src/lib/core/fiber.c b/src/lib/core/fiber.c
index 351f1dfad..cb0de871a 100644
--- a/src/lib/core/fiber.c
+++ b/src/lib/core/fiber.c
@@ -258,12 +258,8 @@ pthread_t main_thread_id;
 static size_t page_size;
 static int stack_direction;
 
-#if FIBER_STACK_SIZE < 524288
-/*
- * See the rationale for the minimal fiber stack size in
- * https://github.com/tarantool/tarantool/issues/3418.
- */
-#error "Minimal fiber stack size is 512Kb"
+#ifndef FIBER_STACK_SIZE
+#error "Fiber stack size is not set"
 #endif
 
```

Anyway, I wonder, whether you're OK to remove this check completely from the sources, since we'll still have compilation errors if `FIBER_STACK_SIZE` is not defined, but the diagnostic won't be as much informative as it can be.